src/pages/ImportData.js

```diff
@@ -211,168 +409,8 @@ const ImportData = () => {
-                h(ButtonPrimary, {
-                  style: { marginLeft: '2rem' },
-                  disabled: !selectedTemplateWorkspaceKey,
-                  onClick: () => setIsCloneOpen(true)
-                }, ['Import'])
-              ])
-            ])
-          }],
-          [Utils.DEFAULT, () => {
-            return h(Fragment, [
-              h2({ style: styles.title }, ['Destination of the prepared data']),
-              div({ style: { marginTop: '0.5rem' } }, ['Choose the option below that best suits your needs.']),
-              !userHasBillingProjects && h(linkAccountPrompt),
-              !!filteredTemplates.length && h(ChoiceButton, {
-                onClick: () => setMode('template'),
-                iconName: 'copySolid',
-                title: 'Start with a template',
-                detail: 'Clone from one of our template workspaces that has analyses ready for use'
-              }),
-              h(ChoiceButton, {
-                onClick: () => setMode('existing'),
-                iconName: 'fileSearchSolid',
-                title: 'Start with an existing workspace',
-                detail: 'Select one of your workspaces',
-                disabled: !userHasBillingProjects
-              }),
-              h(ChoiceButton, {
-                onClick: () => setIsCreateOpen(true),
-                iconName: 'plus-circle',
-                title: 'Start with a new workspace',
-                detail: 'Set up an empty workspace that you will configure for analysis',
-                'aria-haspopup': 'dialog',
-                disabled: !userHasBillingProjects
-              }),
-              isCreateOpen && h(NewWorkspaceModal, {
-                requiredAuthDomain: ad,
-                customMessage: isDataset && noteMessage,
-                onDismiss: () => setIsCreateOpen(false),
-                onSuccess: w => {
-                  setMode('existing')
-                  setIsCreateOpen(false)
-                  setSelectedWorkspaceId(w.workspaceId)
-                  refreshWorkspaces()
-                  onImport(w)
-                }
-              })
-            ])
-          }]
-        ),
-        isCloneOpen && h(NewWorkspaceModal, {
-          cloneWorkspace: _.find({ workspace: selectedTemplateWorkspaceKey }, workspaces),
-          title: `Clone ${selectedTemplateWorkspaceKey.name} and Import Data`,
-          buttonText: 'Clone and Import',
-          customMessage: isDataset && noteMessage,
-          onDismiss: () => setIsCloneOpen(false),
-          onSuccess: w => {
-            setMode('existing')
-            setIsCloneOpen(false)
-            setSelectedWorkspaceId(w.workspaceId)
-            refreshWorkspaces()
-            onImport(w)
-          }
-        }),
-        (isImporting || loadingWorkspaces) && spinnerOverlay
-      ])
+      h(ImportDataOverview, { snapshots, snapshotResponses, url, isDataset, header }),
```

Comment:
I think it is much more understandable with these chunks pulled out, but would like a second opinion. I think they could potentially be moved to their own files as well, which might reduce the amount in this file and make it easier to parse through the complexity.